File: python/数组/1331. 数组序号转换.py
from typing import List
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        d = dict()
        i= 1
        for v in sorted(arr):
            if v not in d:
                d[v]=i
                i+=1
        res = []
        for i in arr:
            res.append(d[i])
        return res


s = Solution()
arr = [37,12,28,9,100,56,80,5,12]
print(s.arrayRankTransform(arr))

File: go/array/1331. 数组序号转换.go
package array

import (
	"sort"
)

func arrayRankTransform(arr []int) []int {
	d := make(map[int]int)
	var tmp []int
	copy(tmp, arr)
	sort.Ints(tmp)
	i := 1
	for _, v := range tmp {
		if d[v] == 0 {
			d[v] = i
			i++
		}
	}
	var res []int
	for _, v := range arr {
		res = append(res, d[v])
	}
	return res
}
